hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/BatchInvalidator.java

```diff
@@ -133,10 +133,11 @@ private void createAndSendInvalidations(String mapName, InvalidationQueue invali
     }
 
     private void sendInvalidations(String mapName, List<Invalidation> invalidations) {
+        Invalidation invalidation = new BatchNearCacheInvalidation(mapName, invalidations);
```

Comment:
Good work.

One thing that could be a potential problem (I don't enough about how this is being used), but in the old code, if there are no listener, no BatchNearCacheInvalidation is created, unlike the new code. One way to fix this is to create the BatchNearCacheInvalidation lazily.